### PR TITLE
rbd: remove duplicate _fio_rbd_connect

### PR DESCRIPTION
**Pull request by liupan1111**
Signed-off-by: Pan Liu <pan.liu@istuary.com>

### CODE REVIEW

**Comment by axboe** [comment]
You'll have to explain this in more detail, what problem does the patch solve? On the simpler style side of things, don't do crazy casts. Setup local 'rbd' variable the places where you need to access it.

**Comment by liupan1111** [comment]
Sorry for the confuse:
1) In fio_rbd_setup, it will call _fio_rbd_connect first, and call rbd_stat to get image size, and disconnect no matter get image successful or not. And call _fio_rbd_connect again, in fio_rbd_init.
    If client is already connect to cluster and get image size successful in fio_rbd_setup, we don't need to disconnect and connect again. For each connect operation, it will create nearly 20 work threads of librbd, and destroy them when disconnect.   When fio jobs are many, it is time consuming.

2) I will remove unreasonable casts, Thanks!

**Comment by liupan1111** [comment]
@axboe removed casts, please take a look.

File: engines/rbd.c [hunk 1]
```diff
@@ -529,6 +532,11 @@ static int fio_rbd_queue(struct thread_data *td, struct io_u *io_u)
 static int fio_rbd_init(struct thread_data *td)
 {
 	int r;
+	struct rbd_data *rbd;
+	rbd = (struct rbd_data*)(td->io_ops_data);
```
Comment (axboe) [inline engines/rbd.c]: You don't have to cast it, it's a void pointer. Just do:

struct rbd_data *rbd = td->io_ops->data;

like everywhere else in the code.

File: engines/rbd.c [hunk 2]
```diff
@@ -529,6 +532,11 @@ static int fio_rbd_queue(struct thread_data *td, struct io_u *io_u)
...
+	if(rbd && rbd->connected) {
```
Comment (axboe) [inline engines/rbd.c]: Can 'rbd' ever be NULL here?

Also style - please follow the general style in the file. In this csae, space after the 'if'.

**Comment by axboe** [comment]
Added some review comments.

File: engines/rbd.c (hunk 1 above)
Comment (liupan1111) [inline engines/rbd.c]: cool.. right

File: engines/rbd.c (hunk 2 above)
Comment (liupan1111) [inline engines/rbd.c]: seems impossible to be NULL. Add if(rbd &&...) just want to make code analysis tool clean. Do you prefer to remove this check？

**Comment by liupan1111** [comment]
@axboe  modified as your suggestion. Thanks!

**Comment by axboe** [comment]
I pulled this, but I still had to fix up some of your styling. No braces for single line if's, and always leave an empty line between variables and code.

Really, this goes for ANY project you want to contribute to. Follow the local style, it's important. It makes the code more easily readable and maintainable.

**Comment by liupan1111** [comment]
er... yes, I'm so sorry always leaves you coding style issue...

**Pull request merged**